crates/forge_analyzer/src/definitions.rs

```diff
@@ -819,21 +956,119 @@ impl<'cx> FunctionAnalyzer<'cx> {
+                } else {
+                    None
                 }
             }
+            // import { foo } from 'foo';
+            // foo.bar.sign('what');
+            // foo.bar.bar.sign('whatever');
+            // PackageData { package: 'foo', identifier: 'bar', method: 'sign' }
+
+            // [Def, Static]
+            //[PropPath::Def(def), PropPath::Static(atom)]
+            // 1. Star, identifier (NO METHOD)
+            // 2. Named/Default => import kind matches idenntifier and static(atom) matches method
+            [PropPath::Def(def), PropPath::Static(ref method_name)] => {
+                if let Some((package_name, import_kind)) = self.res.as_foreign_import(def) {
+                    // Star imports that don't have any object call to invoke function
+                    // import * as atlassian_jwt from "atlassian-jwt";
+                    // atlassian_jwt.encode(blah, blah);
+
+                    if import_kind == ImportKind::Star {
+                        let package_found = self.secret_packages.iter().find(|&package_data| {
+                            package_name == package_data.package_name
+                                && *method_name == package_data.identifier
+                                && package_data.method.is_none()
+                        });
+                        if let Some(package) = package_found {
+                            Some(Intrinsic::JWTSign(package.clone()))
+                        } else {
+                            None
+                        }
+                    } else {
+                        // Named or default imports that
+                        // import AES from "crypto-js";
+                        // import {AES} from "crypto-js"
+                        // Aes.encrypt(blah, blah);
+                        let package_found = self.secret_packages.iter().find(|&package_data| {
+                            if let Some(method) = &package_data.method {
+                                return package_name == package_data.package_name
+                                    && import_kind == *package_data.identifier
+                                    && *method_name == *method;
+                            } else {
+                                false
+                            }
+                        });
+                        if let Some(package) = package_found {
+                            Some(Intrinsic::JWTSign(package.clone()))
+                        } else {
+                            None
+                        }
+                    }
+                } else {
+                    None
+                }
+            }
+            // [Def, .., Static] star
+            // [Def, .., Static] named/default
+            // [PropPath::Def(def)]
+            // 1. Named/Default => package matches imported from and ident is the same, (NO METHOD)
+            [PropPath::Def(def)] if self.res.as_foreign_import(def).is_some() => {
```

Comment:
We should be able to remove this guard as long as this match branch happens near the end, since you have to lookup the `package_name` and `import_kind` in the match arm's block using this function anyway.